File: backend/models/History.js
const mongoose = require("mongoose");

const HistorySchema = new mongoose.Schema(
  {
    type: {
      type: Number,
      // 0 = log, 1 = comment
      enum: [0, 1],
      required: true,
    },
    task: {
      type: mongoose.Schema.ObjectId,
      ref: "Task",
      required: true,
    },
    comment: {
      type: mongoose.Schema.ObjectId,
      ref: "Comment",
    },
    log: {
      type: mongoose.Schema.ObjectId,
      ref: "Log",
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model("History", HistorySchema);

File: backend/app/models/history.rb
# == Schema Information
#
# Table name: histories
#
#  id           :bigint           not null, primary key
#  history_type :integer
#  created_at   :datetime         not null
#  updated_at   :datetime         not null
#  task_id      :bigint
#
# Indexes
#
#  index_histories_on_task_id  (task_id)
#
class History < ApplicationRecord
  has_one :log, :dependent => :destroy
  has_one :comment, :dependent => :destroy
  belongs_to :task

  enum :history_type => {
    :log_item => 0,
    :comment_item => 1
  }
end
